stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/attachments/AttachmentsAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package io.getstream.chat.android.ui.message.list.adapter.attachments
+
+import android.view.ViewGroup
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.ui.common.adapters.SimpleListAdapter
+import io.getstream.chat.android.ui.common.extensions.internal.isMedia
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentViewHolderFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentViewHolderFactoryImpl
+
+internal class AttachmentsAdapter(
+    private val viewHolderFactory: AttachmentViewHolderFactory,
+) : SimpleListAdapter<List<Attachment>, SimpleListAdapter.ViewHolder<List<Attachment>>>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<List<Attachment>> {
+        return viewHolderFactory.attachmentViewHolder(parent, viewType)
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return if (itemList.first().isMedia())
+            AttachmentViewHolderFactoryImpl.MEDIA
+        else
+            AttachmentViewHolderFactoryImpl.FILE
+    }
```

Comment:
This should be determined by the `AttachmentViewHolderFactory` as well, so that clients can override the logic